### PR TITLE
ENH : ability to specify a number of workers

### PR DESCRIPTION
**Pull request by jnrdt**
https://github.com/chrisvfritz/prerender-spa-plugin/issues/53

### CODE REVIEW

**Comment by drewlustro** [comment]
This is ripe for a rewrite in ES6 / ES2015, @jnrdt – what you think?

**Comment by jnrdt** [comment]
@drewlustro yep, I initially did it in es6, but it would need babel.
That would worth adding a babel task don't you think?

File: index.js [hunk 1]
```diff
@@ -11,43 +12,49 @@ function SimpleHtmlPrecompiler (staticDir, paths, options) {
...
-      // setTimeout prevents the Promise from swallowing the throw
-      setTimeout(function () { throw error })
     })
+  }
+  
+  compiler.plugin('after-emit', function (compilation, done) {
+	
+    _.chain(self.paths)
+      .groupBy(function (val) { return _.indexOf(self.paths, val) % waves})
+      .reduce(function (memo, tab, indx) { return memo.then(function () { return Promise.all(tab.map(function (route) { return prerender(route)})) }) }, new Promise(function (resolve) { return resolve() }))
```
Comment (lyzs90) [inline index.js]: this line is kinda long, could use some formatting

File: index.js [hunk 2]
```diff
@@ -11,43 +12,49 @@ function SimpleHtmlPrecompiler (staticDir, paths, options) {
...
-    .then(function () { done() })
-    .catch(function (error) {
-      // setTimeout prevents the Promise from swallowing the throw
-      setTimeout(function () { throw error })
     })
+  }
+  
+  compiler.plugin('after-emit', function (compilation, done) {
+	
+    _.chain(self.paths)
```
Comment (lyzs90) [inline index.js]: can probably do an implicit chain

**Comment by Tribex** [comment]
Thanks for your effort! We now have [support for limiting the maximum number of pages being rendered concurrently](https://github.com/chrisvfritz/prerender-spa-plugin/tree/v3#prerendererrenderer-puppeteer-options) in v3. Since v2 isn't really under active maintenance anymore, I'm going to go ahead and close this.

**Pull request closed without merging**